Engine/src/Engine/UserInterface.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <map>
+#include <string>
+
+#include "imgui.h"
+#include "imgui_impl_opengl3.h"
+#include "imgui_impl_glfw.h"
+
+#include "glm/glm.hpp"
+
+class UserInterface
+{
+
+public:
+
+	//initializes ImGUI vec4s to be used for colours
+	static ImVec4 red;
+	static ImVec4 green;
+	static ImVec4 blue;
+	static ImVec4 white;
+
+	//initialize start and end functions
+	static void startUI();
+	static void endUI();
+
+	//loadFont takes in a string file name, float font size, and string tag and returns true if the font is valid
+	static bool loadFont(const std::string& fontFilename, float fontSize, const std::string& tag);
+
+	virtual ~UserInterface() = 0 {}
+
+	//initialize takes in a GLFWwindow as a parameter and initializes a window to be used with ImGUI
+	static bool initialize(GLFWwindow* window);
+
+	//set position takes in a vec2 and is used to set the position of the window
+	void setPosition(const glm::uvec2& position);
+	void setDimension(const glm::uvec2& dimension);
+	void setButtonDimension(const glm::uvec2& dimension);
+
+	void spacing(int amount);
```

Comment:
"spacing" just adds space around a particular UI element?

Reply:
@Snow106 updated this to setSpacing